step-templates/sql-deploy-dacpac-sqlpackage.json

```diff
@@ -0,0 +1,179 @@
g -ServerName $DACPACTargetServer -Database $DACPACTargetDatabase -UserName $UserName -Password $Password -AuthenticationType $AuthenticationType)`\"\"\n    \n\t# Check to see if a publish profile was designated\n\tIf ($PublishProfile){\n    \t$PublishProfilePath = Get-DacFilePath -BasePath $ExtractPath -FileName $PublishProfile -FileType \"PublishProfile\"\n    \n    \t# Add to arguments\n    \t$sqlPackageArguments += \"/Profile:`\"$PublishProfilePath`\"\"\n\t}    \n    \n    # Check to see if it's using managed identity\n    if ($authenticationType -eq \"AzureADManaged\")\n    {\n    \t# Add access token\n        $Password = Get-ManagedIdentityToken\n        $sqlPackageArguments += \"/AccessToken:$Password\"\n    }\n    \n    # Add sqlcmd variables\n    $sqlPackageArguments += Add-SqlCmdVariables\n    \n\t# Add addtional arguments\n    $sqlPackageArguments += Add-AdditionalArguments -AdditionalArguments $DACPACAdditionalArguments\n    \n    # Check to see if sqlpackage needs to be downloaded\n    if ([string]::IsNullOrWhitespace($DACPACExeLocation))\n    {\n    \t# Download and extract sqlpackage\n        Get-SqlPackage\n    }\n    else\n    {\n    \t# Add folder location to path\n        $env:PATH = \"$([IO.Path]::GetDirectoryName($DACPACExeLocation))$([IO.Path]::PathSeparator)\" + $env:PATH\n        Write-Host \"It is $($env:PATH)\"\n    }\n    \n    # Execute the actions\n    if ($DACPACReport)\n    {\n    \t$workFolder = \"$($OctopusParameters['Octopus.Action.Package[DACPACPackage].ExtractedPath'])/reports\"\n        $sqlReportArguments = @()\n        $sqlReportArguments += \"/OutputPath:$workFolder/report.xml\"\n        \n        # Validate the folder\n        Validate-Folder -TestPath $workFolder\n        \n        # Execute the action\n        Invoke-SqlPackage -Action \"DeployReport\" -Arguments ($sqlPackageArguments + $sqlReportArguments)\n        \n        # Attach artifacts\n        foreach ($item in (Get-ChildItem -Path $workFolder))\n        {\n        \t# Upload artifact\n            New-OctopusArtifact $item.FullName\n        }\n    }\n    \n    if ($DACPACScript)\n    {\n    \t$workFolder = \"$($OctopusParameters['Octopus.Action.Package[DACPACPackage].ExtractedPath'])/scripts\"\n        $sqlScriptArguments = @()\n        $sqlScriptArguments += \"/OutputPath:$workFolder/scripts.sql\"\n        \n        # Validate folder\n        Validate-Folder -TestPath $workFolder\n        \n        # Execute the action\n        Invoke-SqlPackage -Action \"Script\" -Arguments ($sqlPackageArguments + $sqlScriptArguments)\n        \n        # Attach artifacts\n        foreach ($item in (Get-ChildItem -Path $workFolder))\n        {\n        \t# Upload artifact\n            New-OctopusArtifact $item.FullName\n        }        \n    }\n    \n    if ($DACPACDeploy)\n    {\n    \t# Execute action\n        Invoke-SqlPackage -Action \"Publish\" -Arguments $sqlPackageArguments\n    }\n}\ncatch\n{\n    Write-Host $_.Exception.ToString()\n    throw;\n}\n",
```

Comment:
- Line 373: `#Write-Host "Executing `"sqlpackage $Arguments `""` can be removed
- Line 379: `$arrayIndex = 0` value is assigned but not used, so assume it can be removed
- Lines 400-414 (commented code about error redirection) can be removed
-

Reply:
Addressed requested edits.